File: csharp/MapleStory.OpenAPI/Src/Dto/Character/CharacterSymbolEquipmentInfoDTO.cs
using Newtonsoft.Json;

namespace MapleStory.OpenAPI.Dto
{
    /// <summary>
    /// 캐릭터 심볼 상세 정보
    /// </summary>
    public class CharacterSymbolEquipmentInfoDTO
    {
        /// <summary>
        /// 심볼 명
        /// </summary>
        [JsonProperty("symbol_name")]
        public string SymbolName { get; set; }

        /// <summary>
        /// 심볼 아이콘
        /// </summary>
        [JsonProperty("symbol_icon")]
        public string SymbolIcon { get; set; }

        /// <summary>
        /// 심볼 설명
        /// </summary>
        [JsonProperty("symbol_description")]
        public string SymbolDescription { get; set; }

        /// <summary>
        /// 심볼로 인한 증가 수치
        /// </summary>
        [JsonProperty("symbol_force")]
        public string SymbolForce { get; set; }

        /// <summary>
        /// 심볼 레벨
        /// </summary>
        [JsonProperty("symbol_level")]
        public long SymbolLevel { get; set; }

        /// <summary>
        /// 심볼로 증가한 힘
        /// </summary>
        [JsonProperty("symbol_str")]
        public string SymbolStr { get; set; }

        /// <summary>
        /// 심볼로 증가한 민첩
        /// </summary>
        [JsonProperty("symbol_dex")]
        public string SymbolDex { get; set; }

        /// <summary>
        /// 심볼로 증가한 지력
        /// </summary>
        [JsonProperty("symbol_int")]
        public string SymbolInt { get; set; }

        /// <summary>
        /// 심볼로 증가한 운
        /// </summary>
        [JsonProperty("symbol_luk")]
        public string SymbolLuk { get; set; }

        /// <summary>
        /// 심볼로 증가한 체력
        /// </summary>
        [JsonProperty("symbol_hp")]
        public string SymbolHp { get; set; }

        /// <summary>
        /// 심볼로 증가한 아이템 드롭률
        /// </summary>
        public string SymbolDropRate
        {
            get
            {
                return _symbolDropRate == null ? "0%" : _symbolDropRate;
            }
        }

        [JsonProperty("symbol_drop_rate")]
        private string _symbolDropRate { get; set; }

        /// <summary>
        /// 심볼로 증가한 체력
        /// </summary>
        public string SymbolMesoRate
        {
            get
            {
                return _symbolMesoRate == null ? "0%" : _symbolMesoRate;
            }
        }

        [JsonProperty("symbol_meso_rate")]
        private string _symbolMesoRate { get; set; }

        /// <summary>
        /// 심볼로 증가한 체력
        /// </summary>
        public string SymbolExpRate
        {
            get
            {
                return _symbolExpRate == null ? "0%" : _symbolExpRate;
            }
        }

        [JsonProperty("symbol_exp_rate")]
        private string _symbolExpRate { get; set; }

        /// <summary>
        /// 현재 보유 성장치
        /// </summary>
        [JsonProperty("symbol_growth_count")]
        public long SymbolGrowthCount { get; set; }

        /// <summary>
        /// 성장 시 필요한 성장치
        /// </summary>
        [JsonProperty("symbol_require_growth_count")]
        public long SymbolRequireGrowthCount { get; set; }
    }
}

File: java/src/main/java/dev/spiralmoon/maplestory/api/msea/dto/character/CharacterSymbolEquipmentInfoDTO.java
package dev.spiralmoon.maplestory.api.msea.dto.character;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

/**
 * Symbol information
 */
@AllArgsConstructor
@NoArgsConstructor
@Data
@ToString
public class CharacterSymbolEquipmentInfoDTO implements dev.spiralmoon.maplestory.api.common.dto.character.CharacterSymbolEquipmentInfoDTO {

    /**
     * Symbol name
     */
    @JsonProperty("symbol_name")
    private String symbolName;

    /**
     * Symbol icon
     */
    @JsonProperty("symbol_icon")
    private String symbolIcon;

    /**
     * Symbol description
     */
    @JsonProperty("symbol_description")
    private String symbolDescription;

    /**
     * Increase in stats due to the symbol
     */
    @JsonProperty("symbol_force")
    private String symbolForce;

    /**
     * Symbol level
     */
    @JsonProperty("symbol_level")
    private long symbolLevel;

    /**
     * Increase in Strength due to the symbol
     */
    @JsonProperty("symbol_str")
    private String symbolStr;

    /**
     * Increase in Dexterity due to the symbol
     */
    @JsonProperty("symbol_dex")
    private String symbolDex;

    /**
     * Increase in Intelligence due to the symbol
     */
    @JsonProperty("symbol_int")
    private String symbolInt;

    /**
     * Increase in Luck due to the symbol
     */
    @JsonProperty("symbol_luk")
    private String symbolLuk;

    /**
     * Increase in HP due to the symbol
     */
    @JsonProperty("symbol_hp")
    private String symbolHp;

    /**
     * Increase in item drop rate due to the symbol
     */
    @JsonProperty("symbol_drop_rate")
    private String symbolDropRate;

    /**
     * Increase in Meso acquisition rate due to the symbol
     */
    @JsonProperty("symbol_meso_rate")
    private String symbolMesoRate;

    /**
     * Increase in EXP acquisition rate due to the symbol
     */
    @JsonProperty("symbol_exp_rate")
    private String symbolExpRate;

    /**
     * Current growth points
     */
    @JsonProperty("symbol_growth_count")
    private long symbolGrowthCount;

    /**
     * Growth points required for the next level
     */
    @JsonProperty("symbol_require_growth_count")
    private long symbolRequireGrowthCount;

    /**
     * Increase in item drop rate due to the symbol
     */
    public String getSymbolDropRate() {
        return symbolDropRate != null
                ? symbolDropRate
                : "0%";
    }

    /**
     * Increase in Meso acquisition rate due to the symbol
     */
    public String getSymbolMesoRate() {
        return symbolMesoRate != null
                ? symbolMesoRate
                : "0%";
    }

    /**
     * Increase in EXP acquisition rate due to the symbol
     */
    public String getSymbolExpRate() {
        return symbolExpRate != null
                ? symbolExpRate
                : "0%";
    }
}
